File: SilksyAPI/Entities/Address.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace SilksyAPI.Entities
{
    public class Address
    {
        [Key]
        public int AddressId { get; set; }

        [Required]
        [MaxLength(200)]
        public string AddressLine1 { get; set; }

        [Required]
        [MaxLength(200)]
        public string AddressLine2 { get; set; }

        [Required]
        [MaxLength(100)]
        public string City { get; set; }

        [Required]
        [MaxLength(10)]
        public string Postcode { get; set; }

        [Required]
        public DateTime LastUpdate { get; set; }

        public ICollection<UserAddress> UserAddresses { get; set; }
    }
}

File: SilksyClient/src/app/model/address.ts
export class Address {

    constructor(addressLine1, addressLine2, country, city, postCode) {
        this.addressLine1 = addressLine1;
        this.addressLine2 = addressLine2;
        this.country = country;
        this.city = city;
        this.postcode = postCode;
    }

    addressLine1: string;
    addressLine2: string;
    city: string;
    country: string;
    postcode: string;
}